docs/lab/06/index.md

```diff
@@ -0,0 +1,359 @@
+---
+sidebar_position: 7
+description: Serial Peripheral Interface
+slug: /lab/06
+---
+
+# 06 - Serial Peripheral Interface
+
+This lab will teach you how to communicate with hardware devices using the Serial Peripheral Interface (SPI) protocol, in Embassy. 
+
+## Resources
+
+TBD
+
+## Serial Peripheral Interface (SPI)
+
+The Serial Peripheral Interface (or SPI) is a communication protocol used between microcontrollers and peripheral ICs (integrated circuits). It allows synchronous transfer of data between one *main* device and one or more *sub* devices (e.g. sensors). The main device is the one that dictates the communication.
+
+### Data transmission
+
+![SPI_protocol](images/spi_network.svg)
+
+During SPI transmission, there are 4 wires used:
+
+- **CLK** - clock - generated by the main device - subs use it to synchronously read and write to the bus
+- **MOSI** - **M**ain **O**ut **S**ub **I**n - used to carry data from the main device to *all* the subs
+- **MISO** - **M**ain **I**n **S**ub **O**ut - used to carry data from the *active* sub to the main device
+- **CS** - chip select - selects the sub to be activated, or the sub that the main wants to communicate with
+                       - one wire per sub
+                       - inactive subs will disconnect from the *MOSI* and *MISO* lines
+
+:::info
+The communication is *full-duplex*. This means that, during data transmission, both the master and the slave **must write** to the MOSI/MISO wires, no matter if that data is relevant or not.
```

Comment:
explain why... for every clk pulse dictated by main, 1 bit must be written

Reply:
Done. Is it clear now?